### PR TITLE
Modularize `proto::messsage`

### PR DESCRIPTION
**Pull request by overhacked**
Break up `src/proto/message.rs` into a tree of submodules. Also move a couple of unit tests from `message` into submodules, because they tested only those types' functionality.

This change is motivated by #66, which--in part--fixes a bug made possible by `pub` access to the inner `Vec` of the `Unparsed` newtype. Refactoring the `proto::message` types into modules ensures that the message parsing implementations can't directly access the inner fields of the types unless they are deliberately `pub`. When they were all in the same module, visibility rules didn't matter.

### CODE REVIEW

**Comment by wiktor-k** [comment]
:+1: looks quite a bit more manageable.

Would you be able to run formatting with `cargo +nightly fmt`? (for ordering of imports) and add sign-offs with `git rebase --signoff wiktor/fix-extension-serialize`? (I think that's the base branch you're using) and push that stuff with `git push --force-with-lease`?

Thanks a lot :clap: The files are really smaller now... and it occurred to me that we could move quite some hex-encoded vectors to `tests/messages` which would further simplify the code... but that's for another PR :sweat_smile:

**Review by wiktor-k — APPROVED** [review]
Great! Looks good :)

File: src/proto/message/unparsed.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//! Generic container for [`Extension`](super::Extension)-specific content
+
+use ssh_encoding::{self, Decode, Encode, Writer};
+
+/// Generic container for [`Extension`](super::Extension)-specific content.
+/// Accessing the inner `Vec<u8>` is only possible via conversion methods.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Unparsed(Vec<u8>);
```
Comment (baloo) [inline src/proto/message/unparsed.rs:8]: This effectively reverts https://github.com/wiktor-k/ssh-agent-lib/pull/49

Comment (baloo) [reply src/proto/message/unparsed.rs:8]: furthermore: the `From` implementation allows you to place data in there formatted however you'd like, but you can't get it out :(

Comment (wiktor-k) [reply src/proto/message/unparsed.rs:8]: 🤦‍♂️ indeed. It's my mistake!

I was planning to add conversions from/to vectors and then did some other stuff and forgot about this.

Deeply sorry about it, I'll get this fixed and add some tests with non SSH messages as a way to repent for this blunder. 🙇‍♂️

Comment (baloo) [reply src/proto/message/unparsed.rs:8]: it's all good :) 

I opened https://github.com/wiktor-k/ssh-agent-lib/pull/72, I'd like to hear from both of you there.

**Pull request merged**